evm/src/executor/inspector/tracer.rs

```diff
@@ -88,12 +88,36 @@ impl Tracer {
         trace.trace.steps.push(step);
     }
 
-    pub fn fill_step(&mut self, gas: u64, status: Return) {
+    pub fn fill_step<DB: Database>(
+        &mut self,
+        interp: &mut Interpreter,
+        data: &mut EVMData<'_, DB>,
+        status: Return,
+    ) {
         let (trace_idx, step_idx) =
             self.step_stack.pop().expect("can't fill step without starting a step first");
         let step = &mut self.traces.arena[trace_idx].trace.steps[step_idx];
 
-        step.gas_cost = step.gas - gas;
+        let pc = interp.program_counter() - 1;
+        let op = interp.contract.bytecode.bytecode()[pc];
+
+        let journal_entry = data
+            .journaled_state
+            .journal
+            .last()
+            .expect("not enough contract journal entries")
+            .last()
+            .expect("not enough call journal entries");
```

Comment:
this should always work since, revm initializes this with vec![vec![]] and has an entry at this point